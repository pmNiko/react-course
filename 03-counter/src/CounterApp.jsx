import PropTypes from 'prop-types'
import { useState } from 'react';

export const CounterApp = ({value}) => {
  const [counter, setCounter] = useState(value)

  const handleAdd       = () => setCounter(prev => prev + 1)
  const handleSubstract = () => setCounter(prev => prev - 1)
  const handleReset     = () => setCounter(value)

  return (
    <>
      <h1>Counter App</h1>

      <h2>{counter}</h2>

      <button aria-label="btn-add" onClick={handleAdd} >+1</button>
      <button aria-label="btn-reset" onClick={handleReset} >Reset</button>
      <button aria-label="btn-substract" onClick={handleSubstract} >-1</button>
    </>
  )
}

CounterApp.propTypes = {
  value: PropTypes.number.isRequired
}