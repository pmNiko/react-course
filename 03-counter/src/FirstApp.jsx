import PropTypes from "prop-types";

const FirstApp = ({ title, subtitle = 'Subtitulo de muestra' , name = 'no name' }) => {
  return (
    <>
      <h1 data-testid='test-title' >{title}</h1>

      <p>{subtitle}</p>
      
      <p>{subtitle}</p>
      
      <p>{name}</p>
    </>
  )
}


FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string,
  name: PropTypes.string
}


export default FirstApp;