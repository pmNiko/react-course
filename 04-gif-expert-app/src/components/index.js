export * from './GifGrid'
export * from './GifItem'
export * from './AddCategory'
export * from './RemoveGifButton'