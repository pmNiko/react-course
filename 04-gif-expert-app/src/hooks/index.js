export * from './useFetchGifs'
export * from './useHandleCategories'