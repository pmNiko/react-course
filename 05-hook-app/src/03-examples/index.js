export * from './MultipleCustomHooks.jsx'
export * from './LoadingMessage.jsx'
export * from './PokemonCard.jsx'