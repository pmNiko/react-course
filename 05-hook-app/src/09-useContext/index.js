export * from './HomePage'
export * from './AboutPage'
export * from './LoginPage'
export * from './MainApp'
export * from './NavBar'