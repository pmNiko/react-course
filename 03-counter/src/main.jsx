import React from 'react'
import ReactDOM from 'react-dom/client'
import './styles.css'
import  FirstApp  from './FirstApp';
import { CounterApp } from './CounterApp';


ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <FirstApp title='Hola soy Goku' subtitle='Subtitulo de prueba' />
    {/* <CounterApp value={0} /> */}
  </React.StrictMode>
)
