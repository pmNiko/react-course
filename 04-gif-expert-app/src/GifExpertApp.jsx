import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';
import {useHandleCategories} from './hooks'

export const GifExpertApp = () => {
  const {categories, onAddCategory, onRemoveCategory} = useHandleCategories();

  return (
    <>      
      <h1>GifExpertApp</h1>

      <AddCategory 
        onAddCategory={onAddCategory} 
      />     

      {
        categories.map((category, i) => 
          <GifGrid 
            key={category+i} 
            category={category} 
            onRemoveCategory={() => onRemoveCategory(category)} 
          />
        )
      }
    </>
  )
}