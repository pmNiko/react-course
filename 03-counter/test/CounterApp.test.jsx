import  { fireEvent, render, screen }  from '@testing-library/react'; 
import  { CounterApp }  from '../src/CounterApp';

describe('Pruebas del componente <Counter />', () => { 

  const value = 100;
  const btnAdd = '+1'
  const btnSubstract = '-1'
  const btnReset = 'Reset'
  
  test('Debe hacer match con el snapshot', () => { 
    const { container } = render(<CounterApp value={value} />);

    expect(container).toMatchSnapshot();
  })

  
  test('Debe debe mostrar el valor inicial de 100', () => { 
    const { container } = render(<CounterApp value={value} />);

    expect( screen.getByRole('heading', {level: 2} ).innerHTML ).toContain(value.toString());
  })

  test('Debe de incrementar con el boton +1', () => { 
    render(<CounterApp value={1} />); 

    expect( screen.getByRole('heading', {level: 2}).innerHTML ).toBe('1');

    fireEvent.click( screen.getByText(btnAdd) );

    expect( screen.getByRole('heading', {level: 2}).innerHTML ).toBe('2');
  })

  test('Debe de decrementar con el boton -1', () => { 
    render(<CounterApp value={10} />); 

    expect( screen.getByRole('heading', {level: 2}).innerHTML ).toBe('10');

    fireEvent.click( screen.getByText(btnSubstract) );

    expect( screen.getByRole('heading', {level: 2}).innerHTML ).toBe('9');
  })

  test('Debe regresar a su valor inicial con el boton "Reset"', () => { 
    render(<CounterApp value={0} />); 
    const btnAddName = screen.getByRole('button', {name: 'btn-add'});
    const btnResetName = screen.getByRole('button', {name: 'btn-reset'}); 
    for ( const i of Array(3) ) fireEvent.click( btnAddName );
    
    expect( screen.getByText( 3 ) ).toBeTruthy();
    
    fireEvent.click( btnResetName );

    expect( screen.getByText( 0 ) ).toBeTruthy();
  })
})