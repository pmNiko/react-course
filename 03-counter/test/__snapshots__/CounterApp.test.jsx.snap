// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del componente <Counter /> Debe hacer match con el snapshot 1`] = `
<div>
  <h1>
    Counter App
  </h1>
  <h2>
    100
  </h2>
  <button
    aria-label="btn-add"
  >
    +1
  </button>
  <button
    aria-label="btn-reset"
  >
    Reset
  </button>
  <button
    aria-label="btn-substract"
  >
    -1
  </button>
</div>
`;
